ExcelWriter.py

```diff
@@ -1,130 +1,228 @@
+        for i in range(0, len(lines)):
+            lines[i] = self.String_Manip(lines[i], tabSize)
+        self.entries[-1].append(lines)
         
+
+    #When we are finally creating our output, we want to put solid borders
+    #separating the groups, but we don't want to create a ton of the same
+    #format, so map them by the solid border type, font name & whether the
+    #font will be bold.
+    def Get_Border_Format(self, topBorder, bottomBorder, fontName, isHeader):
+        if ((topBorder, bottomBorder, fontName, isHeader) not in self.borderFormat):
+            temp_format = self.workbook.add_format({'font_name':fontName})
+            temp_format.set_font_size(10)
+            temp_format.set_align('top')
+            temp_format.set_bottom(bottomBorder)
+            temp_format.set_top(topBorder)
+            temp_format.set_bold(isHeader)            
+            if (isHeader):
+                temp_format.set_align('center')
+            self.borderFormat[(topBorder, bottomBorder, fontName, isHeader)] = temp_format
+        return self.borderFormat[(topBorder, bottomBorder, fontName, isHeader)]
+
+    # Determine how big we want the columns to be
+    # Figure out the width of all the columns, sort
+    # them, then chop off the last 3 to kill off the giant
+    # outliers (3 was an arbitrary number that seemed to work)
+    # averge didn't work well, and neither did median (when we
+    # have a lot of missing assignment, the median goes to zero).
+    def Compute_Col_Width(self):
+        maxCol = []
+        colWidths=[[] for i in range(0, len(self.fontNames))]
+        for studentValues in self.entries:
+            currentCol = 0            
+            for entryList in studentValues:
+                for entry in entryList:              
+                    colWidths[currentCol].append(len(entry) + 1)
+                currentCol = currentCol + 1
         
-    def Close(self):
-        #self.Merge_Cells()
-        
+        for colWidthList in colWidths:
+            colWidthList.sort();
+            #print(colWidthList)
+            # not quite the max, trim off the last 3 entries, because some of those
+            # are very large when there is an error 
+            pos = len(colWidthList) - 3
+            if (pos < 0):
+                pos = len(colWidthList) - 1
+            maxCol.append(colWidthList[pos]);        
+        return maxCol
+    
+    # Set the widths of the columns in the spreadsheet based on the
+    # size of the strings
+    def Set_Col_Widths(self):
+        maxCol = self.Compute_Col_Width()
+        #print(maxCol)
         colNum = 0
-        for colWidth in self.maxCol:
+        for colWidth in maxCol:
             colString = self.letters[colNum] + ":" + self.letters[colNum]
             #print ("setting width " + colString + " " + str(colWidth))
             self.worksheet.set_column(colString, colWidth)
```

Comment:
According to [the docs for xlsxwriter](https://xlsxwriter.readthedocs.io/worksheet.html#worksheet-set-column), it seems possible to just call `set_column` with numerical indices so I don't think we need to use `self.letters` here.

Reply:
For some reason, it i didn't work.  I didn't dig deep into it.  I did the "Let me try the other interface, oh that worked" method.